linera-views/src/common.rs

```diff
@@ -138,6 +135,9 @@ pub trait KeyValueStoreClient {
     /// Returns type for key-value search operations.
     type KeyValues: KeyValueIterable<Self::Error>;
 
+    /// Retrieve the number of simultaneous connections
+    fn max_concurrent_queries(&self) -> usize;
```

Comment:
In hindsight, what we mean to return here is merely the number `n` that should be used by `buffer_unordered`. Therefore, I would suggest to rename this function into something very targeted like `preferred_num_concurrent_requests`. IMHO this is the only way to make `n = 1` understandable in some cases (RocksDB and the KVSView).

On the other hand, this PR introduce the idea that a maximum number of concurrent requests could be _enforced_ by certain implementations of the KVSC trait. Because our motivation is mainly testing, I would like to suggest making this new behavior deactivated by default in production. Concretely, the parameter `max_concurrent_queries` stored inside the implementations would be an `option<usize>` and the semaphore probably an option as well.

Concretely, we would have `self.preferred_num_concurrent_requests()` return some formula like `self.max_concurrent_queries.unwrap_or(THE_CURRENT_VALUE)`.

Reply:
Ok, I have done this.